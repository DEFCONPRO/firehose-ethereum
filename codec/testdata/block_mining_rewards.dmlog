FIRE INIT 2.0 geth 1.10.17-dm-stable
FIRE BEGIN_BLOCK 6430278
FIRE FINALIZE_BLOCK 6430278
FIRE BALANCE_CHANGE 0 6212dd88f890fefe0af24d1404d96adf488e4e3b 1fa95450485cef7730f9 1fa95b40a3b6c32930f9 reward_mine_uncle 1
FIRE BALANCE_CHANGE 0 6212dd88f890fefe0af24d1404d96adf488e4e3b 1fa95b40a3b6c32930f9 1fa96230ff1096db30f9 reward_mine_uncle 2
FIRE BALANCE_CHANGE 0 6212dd88f890fefe0af24d1404d96adf488e4e3b 1fa96230ff1096db30f9 1fa97fae834e5a8fb0f9 reward_mine_block 3
FIRE END_BLOCK 6430278 1902 {"header":{"parentHash":"0x30c20db7becee9037208a4753f68673a3c86a5eb1cce785888023f06843920b8","sha3Uncles":"0x6f4d1df8728afa81353eff1c07f9ef3c31003ac6db6cc9e210ac4bee7dfcb40f","miner":"0x6212dd88f890fefe0af24d1404d96adf488e4e3b","stateRoot":"0x39883f641e811c8ce9946ca90db288ec8b2aecf4286510f54ca93c04305bfb74","transactionsRoot":"0x21705295765edfd55972d0efa5eeaa701964cdd6d4ffe9601a1147e3a4da3ec5","receiptsRoot":"0x95ddfaf1465084fa01443d37bbc7bdfbcad8019b6435f3754344e521209eb20f","logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","difficulty":"0x1b1039fe","number":"0x621e46","gasLimit":"0x7a1200","gasUsed":"0x79c4ab","timestamp":"0x5d85f338","extraData":"0xd883010816846765746888676f312e31302e34856c696e7578","mixHash":"0x6d1872896123f9e61ca0c77d353086def98cdfc3162645d06e20edbc1dc60271","nonce":"0x91085908611a0214","hash":"0x736fb90beb572642b589ee15f8d73516d8a41b4b79ee60fe97565203009c00ea"},"uncles":[{"parentHash":"0xe8fcbb0821c3f944c50dafc232ca2c4d5fd865935939713a05eb9a1c3ff22db6","sha3Uncles":"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347","miner":"0x6212dd88f890fefe0af24d1404d96adf488e4e3b","stateRoot":"0x24bffe381be83b9ae8e4d3980fce1924ef53f6445498f8a1630a0075fe6ccc80","transactionsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","receiptsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","difficulty":"0x1b279b0d","number":"0x621e40","gasLimit":"0x7a1200","gasUsed":"0x0","timestamp":"0x5d85f285","extraData":"0xd883010816846765746888676f312e31302e34856c696e7578","mixHash":"0x3ac5f9c94e5abbce20fc3c483bcd04cb4d0b19611fd0ad7e53dd503277137071","nonce":"0x910859084be6cdfe","hash":"0x2c46694b43b6c8e687edaf1b5257aca53fe1935e6d59c0ee7d00a68b801445e4"},{"parentHash":"0xe8fcbb0821c3f944c50dafc232ca2c4d5fd865935939713a05eb9a1c3ff22db6","sha3Uncles":"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347","miner":"0x6212dd88f890fefe0af24d1404d96adf488e4e3b","stateRoot":"0x24bffe381be83b9ae8e4d3980fce1924ef53f6445498f8a1630a0075fe6ccc80","transactionsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","receiptsRoot":"0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421","logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","difficulty":"0x1b279b0d","number":"0x621e40","gasLimit":"0x7a1200","gasUsed":"0x0","timestamp":"0x5d85f285","extraData":"0xd883010816846765746888676f312e31302e34856c696e7578","mixHash":"0x886abed207767b7f7b36ced34d74076095bd3a103b7fed9fed3489704b9399ba","nonce":"0x910859087242b012","hash":"0x1174ab3f74e43afce47d084bb8c3e9ee8cbf1e886b53ef85b4b97a676ba51561"}]}
