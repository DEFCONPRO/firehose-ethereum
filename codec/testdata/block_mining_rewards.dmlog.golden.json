[{
  "hash": "736fb90beb572642b589ee15f8d73516d8a41b4b79ee60fe97565203009c00ea",
  "number": "6430278",
  "size": "1902",
  "header": {
    "parentHash": "30c20db7becee9037208a4753f68673a3c86a5eb1cce785888023f06843920b8",
    "uncleHash": "6f4d1df8728afa81353eff1c07f9ef3c31003ac6db6cc9e210ac4bee7dfcb40f",
    "coinbase": "6212dd88f890fefe0af24d1404d96adf488e4e3b",
    "stateRoot": "39883f641e811c8ce9946ca90db288ec8b2aecf4286510f54ca93c04305bfb74",
    "transactionsRoot": "21705295765edfd55972d0efa5eeaa701964cdd6d4ffe9601a1147e3a4da3ec5",
    "receiptRoot": "95ddfaf1465084fa01443d37bbc7bdfbcad8019b6435f3754344e521209eb20f",
    "logsBloom": "00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
    "difficulty": "1b1039fe",
    "number": "6430278",
    "gasLimit": "8000000",
    "gasUsed": "7980203",
    "timestamp": "2019-09-21T09:54:00Z",
    "extraData": "d883010816846765746888676f312e31302e34856c696e7578",
    "mixHash": "6d1872896123f9e61ca0c77d353086def98cdfc3162645d06e20edbc1dc60271",
    "nonce": "10450700827836940820",
    "hash": "736fb90beb572642b589ee15f8d73516d8a41b4b79ee60fe97565203009c00ea"
  },
  "uncles": [
    {
      "parentHash": "e8fcbb0821c3f944c50dafc232ca2c4d5fd865935939713a05eb9a1c3ff22db6",
      "uncleHash": "1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
      "coinbase": "6212dd88f890fefe0af24d1404d96adf488e4e3b",
      "stateRoot": "24bffe381be83b9ae8e4d3980fce1924ef53f6445498f8a1630a0075fe6ccc80",
      "transactionsRoot": "56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
      "receiptRoot": "56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
      "logsBloom": "00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
      "difficulty": "1b279b0d",
      "number": "6430272",
      "gasLimit": "8000000",
      "timestamp": "2019-09-21T09:51:01Z",
      "extraData": "d883010816846765746888676f312e31302e34856c696e7578",
      "mixHash": "3ac5f9c94e5abbce20fc3c483bcd04cb4d0b19611fd0ad7e53dd503277137071",
      "nonce": "10450700827481263614",
      "hash": "2c46694b43b6c8e687edaf1b5257aca53fe1935e6d59c0ee7d00a68b801445e4"
    },
    {
      "parentHash": "e8fcbb0821c3f944c50dafc232ca2c4d5fd865935939713a05eb9a1c3ff22db6",
      "uncleHash": "1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
      "coinbase": "6212dd88f890fefe0af24d1404d96adf488e4e3b",
      "stateRoot": "24bffe381be83b9ae8e4d3980fce1924ef53f6445498f8a1630a0075fe6ccc80",
      "transactionsRoot": "56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
      "receiptRoot": "56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
      "logsBloom": "00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
      "difficulty": "1b279b0d",
      "number": "6430272",
      "gasLimit": "8000000",
      "timestamp": "2019-09-21T09:51:01Z",
      "extraData": "d883010816846765746888676f312e31302e34856c696e7578",
      "mixHash": "886abed207767b7f7b36ced34d74076095bd3a103b7fed9fed3489704b9399ba",
      "nonce": "10450700828124819474",
      "hash": "1174ab3f74e43afce47d084bb8c3e9ee8cbf1e886b53ef85b4b97a676ba51561"
    }
  ],
  "balanceChanges": [
    {
      "address": "6212dd88f890fefe0af24d1404d96adf488e4e3b",
      "oldValue": "1fa95450485cef7730f9",
      "newValue": "1fa95b40a3b6c32930f9",
      "reason": "REASON_REWARD_MINE_UNCLE",
      "ordinal": "1"
    },
    {
      "address": "6212dd88f890fefe0af24d1404d96adf488e4e3b",
      "oldValue": "1fa95b40a3b6c32930f9",
      "newValue": "1fa96230ff1096db30f9",
      "reason": "REASON_REWARD_MINE_UNCLE",
      "ordinal": "2"
    },
    {
      "address": "6212dd88f890fefe0af24d1404d96adf488e4e3b",
      "oldValue": "1fa96230ff1096db30f9",
      "newValue": "1fa97fae834e5a8fb0f9",
      "reason": "REASON_REWARD_MINE_BLOCK",
      "ordinal": "3"
    }
  ],
  "ver": 3
}
]